mergesort.rb

```diff
@@ -1,25 +1,45 @@
 def mergesort(a)
     # if the array size is 0|1 then the list is considered sorted, return sorted array
+    return a if a.length < 2
     # split the list into two halves
+    left,right = split_array(a)
     # merge sort each half
+    left = mergesort(left)
+    right = mergesort(right)
     # combine the sorted halves
+    combine(left,right)
 end
 
 def split_array(a)
+  split = a.each_slice((a.length/2.0).round).to_a
+  # probs don't need it but meh)
```

Comment:
There are "code golf" competitions for doing things in the fewest lines of code, if you do enjoy that sort of thing. :-)